orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -57,175 +58,193 @@ def forward(self, *inputs, **kwargs):
-                                                self._graph_info.user_input_names,
-                                                self._input_info,
-                                                self._flattened_module.named_buffers(),
-                                                inputs,
-                                                kwargs,
-                                                self._device)))
+                            # input_name is not found in the self._input_info.require_grad_names list
+                            # Append None to results for each input that did not require grad
+                            results.append(None)
+
+                    # Append gradients of initializer to results
+                    # Go over each initializer, check if it required grad and append to results accordingly
+                    initializer_index = num_user_input_grads
+                    for initializer_name in self._graph_info.initializer_names:
+                        if initializer_name in self._graph_initializer_names_to_train:
+                            results.append(_utils._torch_tensor_from_dl_pack(
+                                backward_outputs.dlpack_at(initializer_index),
+                                backward_outputs[initializer_index]))
+                            initializer_index += 1
+                        else:
+                            results.append(None)
+
+                    return tuple(results)
+
+            return _io.unflatten_user_output(self._module_output_schema,
+                                             _ORTModuleFunction.apply(
+                                                 *_io._combine_input_buffers_initializers(
+                                                     self._graph_initializers,
+                                                     self._graph_info.user_input_names,
+                                                     self._input_info,
+                                                     self._flattened_module.named_buffers(),
+                                                     inputs,
+                                                     kwargs,
+                                                     self._device)))
+        except ORTModuleFallbackException as e:
+            # Exceptions subject to fallback are handled here
+            self._fallback_manager.handle_exception(e)
+        except Exception as e:
+            # Catch-all FALLBACK_FORCE_TORCH_FORWARD fallback is handled here
+            self._fallback_manager.handle_exception(e, _FallbackPolicy.FALLBACK_FORCE_TORCH_FORWARD)
+
+        # Fallback to PyTorch due to failures *during* forward(),
+        #  (e.g. export, model/input post-processing, forward, output processing, etc)
+        if self._fallback_manager.is_pending():
+            return self._fallback_manager.fallback(self._original_module, *inputs, **kwargs)
```

Comment:
Need to test on some real models.
I wonder if backend would need to do some cleanup in such case. (Or maybe it has been handled properly already) 

For example, we might need to clean up the execution frame to free the GPU memory. 
Do we need to delete the ctx? 

We should clean up ORTModule to a state where it becomes light-weighted for pytorch module to take over the remaining execution.

Reply:
That is a good experiment. We should try to run large models that fail to export or something like that.
The retry strategy must be addressed soon too